src/Riok.Mapperly/Descriptors/MappingBodyBuilder/NewInstanceObjectPropertyMappingBodyBuilder.cs

```diff
@@ -195,42 +195,88 @@ private static void BuildInitOnlyPropertyMappings(NewInstanceMappingBuilderConte
         foreach (var parameter in ctor.Parameters)
-        {
-            if (!PropertyPath.TryFind(
-                ctx.Mapping.SourceType,
-                MemberPathCandidateBuilder.BuildMemberPathCandidates(parameter.Name),
-                ctx.IgnoredSourcePropertyNames,
-                StringComparer.OrdinalIgnoreCase,
-                out var sourcePath))
-            {
-                if (!parameter.IsOptional)
-                    return false;
-
-                skippedOptionalParam = true;
-                continue;
-            }
-
-            // nullability is handled inside the property mapping
-            var paramType = parameter.Type.WithNullableAnnotation(parameter.NullableAnnotation);
-            var delegateMapping = ctx.BuilderContext.FindMapping(sourcePath.MemberType, paramType)
-                ?? ctx.BuilderContext.FindOrBuildMapping(sourcePath.Member.Type.NonNullable(), paramType);
-            if (delegateMapping == null)
-            {
-                if (!parameter.IsOptional)
-                    return false;
-
-                skippedOptionalParam = true;
-                continue;
-            }
-
-            var propertyMapping = new NullPropertyMapping(delegateMapping, sourcePath, ctx.BuilderContext.GetNullFallbackValue(paramType));
-            var ctorMapping = new ConstructorParameterMapping(parameter, propertyMapping, skippedOptionalParam);
-            constructorParameterMappings.Add(ctorMapping);
+        {
+            if (!TryFindConstructorParameterSourcePath(ctx, parameter, out var sourcePath))
+            {
+                if (!parameter.IsOptional)
+                    return false;
+
+                skippedOptionalParam = true;
+                continue;
+            }
+
+            // nullability is handled inside the property mapping
+            var paramType = parameter.Type.WithNullableAnnotation(parameter.NullableAnnotation);
+            var delegateMapping = ctx.BuilderContext.FindMapping(sourcePath.MemberType, paramType)
+                ?? ctx.BuilderContext.FindOrBuildMapping(sourcePath.Member.Type.NonNullable(), paramType);
+            if (delegateMapping == null)
+            {
+                if (!parameter.IsOptional)
+                    return false;
+
+                skippedOptionalParam = true;
+                continue;
+            }
+
+            var propertyMapping = new NullPropertyMapping(delegateMapping, sourcePath, ctx.BuilderContext.GetNullFallbackValue(paramType));
+            var ctorMapping = new ConstructorParameterMapping(parameter, propertyMapping, skippedOptionalParam);
+            constructorParameterMappings.Add(ctorMapping);
             mappedTargetPropertyNames.Add(parameter.Name);
         }
 
         return true;
     }
+
+    private static bool TryFindConstructorParameterSourcePath(NewInstanceMappingBuilderContext ctx, IParameterSymbol parameter, [NotNullWhen(true)] out PropertyPath? sourcePath)

```

Comment:
IMO each parameter on a new line could improve readability of this method signature a lot.